### PR TITLE
Add OpenTelemetry extension

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: docs/modules/ROOT/pages/reference/extensions/opentelemetry.adoc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+|icon:lock[title=Fixed at build time] [[quarkus.camel.opentelemetry.encoding]]`link:#quarkus.camel.opentelemetry.encoding[quarkus.camel.opentelemetry.encoding]`
+
+Sets whether header keys need to be encoded
+| `boolean`
+| `false`
+
+|icon:lock[title=Fixed at build time] [[quarkus.camel.opentelemetry.exclude-patterns]]`link:#quarkus.camel.opentelemetry.exclude-patterns[quarkus.camel.opentelemetry.exclude-patterns]`
+
+Sets whether to disable tracing for endpoint URIs that match the given patterns
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/extensions/opentelemetry.adoc:107]: Are they globs or regular expressions? I think as a reader I'd appreciate more clarity.

**Review by ppalaga — APPROVED** [review]
Nice, thanks @jamesnetherton !

File: docs/modules/ROOT/pages/reference/extensions/opentelemetry.adoc (hunk 1 above)
Comment (ppalaga) [reply docs/modules/ROOT/pages/reference/extensions/opentelemetry.adoc:107]: ... and if they are globs, like they seem to based on the example above, what's the syntax? I anything beyond `*` supported?

Comment (jamesnetherton) [reply docs/modules/ROOT/pages/reference/extensions/opentelemetry.adoc:107]: Effectively these are the match rules:

https://github.com/apache/camel/blob/aa262669106bf21ce8d6b6c5e06608e300fa1ea2/core/camel-support/src/main/java/org/apache/camel/support/EndpointHelper.java#L205-L208

I can add some more notes to the config documentation.

**Review by aldettinger — APPROVED** [review]
Looks good to me. As far as I've dug, the encoding option does involve any charsets need at runtime.

**Comment by jamesnetherton** [comment]
Latest commit adds some extra notes to the docs about `exclude-patterns`.

**Pull request merged**